### ISSUE TITLE
Creating default FederationConfig fails

### ISSUE THREAD

**Issue by pmorie** [kubernetes-sigs/federation-v2#816]
**What happened**:

TLDR: the controller does not correctly create a default `FederationConfig` if one does not exist.

We ran the federation controller thusly:

```
$ oc logs -n federation-test federation-controller-manager-7677c44864-44d52
Federation v2 controller-manager version: version.Info{Version:"v0.0.9", GitCommit:"442e8868c6bd02bb61871af02269cb18a58804d4", GitTreeState:"clean", BuildDate:"2019-04-30T05:36:16Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
I0430 21:27:02.844168       1 controller-manager.go:345] FLAG: --alsologtostderr="false"
I0430 21:27:02.844218       1 controller-manager.go:345] FLAG: --federation-config=""
I0430 21:27:02.844226       1 controller-manager.go:345] FLAG: --federation-namespace="federation-test"
I0430 21:27:02.844233       1 controller-manager.go:345] FLAG: --help="false"
I0430 21:27:02.844241       1 controller-manager.go:345] FLAG: --kubeconfig=""
I0430 21:27:02.844247       1 controller-manager.go:345] FLAG: --log-backtrace-at=":0"
I0430 21:27:02.844255       1 controller-manager.go:345] FLAG: --log-dir=""
I0430 21:27:02.844261       1 controller-manager.go:345] FLAG: --log-flush-frequency="5s"
I0430 21:27:02.844270       1 controller-manager.go:345] FLAG: --log_backtrace_at=":0"
I0430 21:27:02.844276       1 controller-manager.go:345] FLAG: --log_dir=""
I0430 21:27:02.844281       1 controller-manager.go:345] FLAG: --logtostderr="true"
I0430 21:27:02.844286       1 controller-manager.go:345] FLAG: --master=""
I0430 21:27:02.844292       1 controller-manager.go:345] FLAG: --stderrthreshold="2"
I0430 21:27:02.844297       1 controller-manager.go:345] FLAG: --v="5"
I0430 21:27:02.844302       1 controller-manager.go:345] FLAG: --version="false"
I0430 21:27:02.844307       1 controller-manager.go:345] FLAG: --vmodule=""
W0430 21:27:02.844424       1 client_config.go:549] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.                                                            
I0430 21:27:03.242267       1 controller-manager.go:198] Cannot retrieve FederationConfig "federation-test/federation-v2": federationconfigs.core.federation.k8s.io "federation-v2" not found. Default options are
used.
F0430 21:27:03.479683       1 controller-manager.go:284] Error retrieving FederationConfig "": resource name may not be empty                              
```

Notice that in the last line of the log, the error message states `resource name may not be empty` - this is because the flow in setOptionsByFederationConfig does not give the default `FederationConfig` a name

**What you expected to happen**:

The controller should create a `FederationConfig` named `federation-v2`.

**How to reproduce it (as minimally and precisely as possible)**:

Run the controller without creating a `FederationConfig`.

**Anything else we need to know?**:

**Environment**:

- Federation version: 0.0.9

/kind bug
